packages/lyra/src/lyra.ts

```diff
@@ -4,12 +4,28 @@ import * as fastq from "fastq";
 import { Trie } from "./prefix-tree/trie";
 import * as ERRORS from "./errors";
 import { tokenize } from "./tokenizer";
-import { formatNanoseconds, getNanosecondsTime } from "./utils";
+import {
+  binarySearch,
+  formatNanoseconds,
+  getNanosecondsTime,
+  setIntersection,
+  setSubtraction,
+} from "./utils";
 import { Language, SUPPORTED_LANGUAGES } from "./stemmer";
-import type { ResolveSchema, SearchProperties } from "./types";
+import type {
+  Nullable,
+  NumberComparison,
+  ResolveSchema,
+  SearchProperties,
+  WhereParams,
+} from "./types";
+import { SortedQueue } from "./sorted-queue";
+import { QueueNode } from "./sorted-queue/node";
 
 export type PropertyType = "string" | "number" | "boolean";
 
+export const allowedNumericComparison = [">=", ">", "=", "<", "<="] as const;
```

Comment:
Can I ask why a symbolic strategy was chosen instead of a literal one? So, for instance: `gte`, `gt`, `eq`, `lt`, `lte` etc.

If clauses like "includes," "between," and so on are introduced in the future, everything will be more consistent. What are your thoughts?

Reply:
@micheleriva Suggested that in the issue. I prefer the symbolic one since I am mostly used to it, but we can start a discussion on it.